### ISSUE TITLE
Posts from spaces showing up on profiles

### ISSUE THREAD

**Issue by WASasquatch** [humhub/humhub#916]
Posts are getting... confused? This post, on a space, is showing up in a new users profile. GUIDs for users and spaces are different so I don't know why HumHub is doing this.

<img src="http://img.prntscr.com/img?url=http://i.imgur.com/W3heqw2.jpg" />
<img src="http://img.prntscr.com/img?url=http://i.imgur.com/l6ihq6s.jpg" />

Space:
<img src="http://img.prntscr.com/img?url=http://i.imgur.com/KBohKEk.jpg" />
<img src="http://img.prntscr.com/img?url=http://i.imgur.com/IueU4Zw.jpg" />

Anyway to fix this without... again.... loosing all my content and starting from scratch for the 4th time @luke- ? No edits to any source, only replaced views. Same case on default HumHub theme. :-1:

**Comment by WASasquatch** [comment]
Anyone? Pretty buggy issue.

**Comment by WASasquatch** [comment]
Any thoughts @luke- @andystrobel this is a severely crippling problem and am close to abandoning HumHub software for another framework. It's just one thing after another, and this seems to be from the base code. There is no error, just how HumHub 0.11.1 is running. So far 0.11.x has proven to be **unstable** and all these errors we're all getting need to be resolved ASAP.

**Comment by luke-** [comment]
@WASasquatch Can you please send a database dump of the corrupted DB to info@humhub.com ? I'll take a look - but it's really a strange behavior...

**Comment by WASasquatch** [comment]
Yes I'll forward a export immediately. I've been trying to figure it out.
Like I've mentioned, for some reason, on my server, even on a fresh install
of 0.10.1 it will say it's up to date. I never can run the DB migration
tool.

On Fri, Apr 10, 2015 at 1:06 PM, Lucas Bartholemy <notifications@github.com>
wrote:

> @WASasquatch <https://github.com/WASasquatch> Can you please send a
> database dump of the corrupted DB to info@humhub.com ? I'll take a look -
> but it's really a strange behavior...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/humhub/humhub/issues/916#issuecomment-91667822>.
>



-- 
Sincerely,
*Jordan S. C. Thompson *
*Web Development & Design Coordinator*

**Comment by luke-** [comment]
I've just imported your database in a clean HumHub 0.11.1 
I can't find any user profile with this problem. Can you give me an username for example?

Btw. the given database is already migrated to 0.11

**Comment by WASasquatch** [comment]
I am WASasquatch, Shaggy @Herbsy.

**Comment by WASasquatch** [comment]
Users in question can be seen in the screenshots.

On Fri, Apr 10, 2015 at 3:57 PM, Jordan Thompson <jordanslost@gmail.com>
wrote:

> I am WASasquatch, Shaggy @Herbsy.
>



-- 
Sincerely,
*Jordan S. C. Thompson *
*Web Development & Design Coordinator*

**Comment by luke-** [comment]
There is no user e.g. "adamcalam" in database.

All other user profiles are empty or showing only user content.
Are you sure this problem is not related to the theme or a code change/custom module?

You can easily try this by downloading the 0.11.1 and testing your database against it.

**Comment by WASasquatch** [comment]
I've done this 4 times now. here is the 5th time. new HumHub 0.11.1 installation.

<img src="http://img.prntscr.com/img?url=http://i.imgur.com/CMOuULx.png" />

**Comment by luke-** [comment]
@WASasquatch I've just send you an email with access to a test server (Vanilla 0.11.1 & Your Test DB) - I hope we can reproduce the problem there...

**Comment by WASasquatch** [comment]
Thanks, I have looked into it, and apparently the DB never was overwritten
on my filesystem from sFTP so that database is the old one. I sent a new
copy, sorry for the inconvenience.

**Comment by luke-** [comment]
Did you replaced/modified something manually in database?

Normally each user and space should have one entry in wall table.
Your wall table only has 6 entries.   Correct should be 44 user + 17 spaces = 61

I'm quite sure this can't happen without manipulating the database manually.

For each user we're creating a wall and store the wall_id in User Record.
https://github.com/humhub/humhub/blob/master/protected/modules_core/user/models/User.php#L404-L412

The user/space and wall table are not belong together in your dump.

**Comment by WASasquatch** [comment]
Remember the activity issues and stuff? I had to drop ALL content besides
users. So IDs are not adding up I would imagine is the issue then.

On Sun, Apr 12, 2015 at 11:02 AM, Lucas Bartholemy <notifications@github.com
> wrote:

> Did you replaced/modified something manually in database?
>
> Normally each user and space should have one entry in wall table.
> Your wall table only has 6 entries. Correct should be 44 user + 17 spaces
> = 61
>
> I'm quite sure this can't happen without manipulating the database
> manually.
>
> For each user we're creating a wall and store the wall_id in User Record.
>
> https://github.com/humhub/humhub/blob/master/protected/modules_core/user/models/User.php#L404-L412
>
> The user/space and wall table are not belong together in your dump.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/humhub/humhub/issues/916#issuecomment-92096585>.
>



-- 
Sincerely,
*Jordan S. C. Thompson *
*Web Development & Design Coordinator*

**Comment by luke-** [comment]
Hmm, I see two options, restore the DB or try to fix it manually.

With following _untested_ script you can try to rebuild the wall table.
It could be possible that meanwhile a space and user has the same wall_id.
I'm not sure how to fix that without loosing content. 


```
        Wall::model()->deleteAll();
        foreach (User::model()->findAll() as $u) {
            if ($u->wall_id != "") {
                $wall = new Wall();
                $wall->id = $u->wall_id;
                $wall->object_model = 'User';
                $wall->object_id = $u->id;
                $wall->save();
            }
        }

        foreach (Space::model()->findAll() as $s) {
            if ($s->wall_id != "") {
                
                if (Wall::model()->findByPk($s->wall_id)) {
                    print "WARN: Duplicate wall_id ".$s->wall_id."for User/Space - Fix manually!\n";
                    continue;
                }
                
                $wall = new Wall();
                $wall->id = $s->wall_id;
                $wall->object_model = 'Space';
                $wall->object_id = $s->id;
                $wall->save();
            }
        }

        // Now set autoincrement of Wall Table manually to max. id
        
```

**Comment by WASasquatch** [comment]
Would importing spaces/users into fresh DB without deleted entries be fresh enough as well?

**Comment by luke-** [comment]
What do you mean with importing?

Most tables are connected with other tables, like user -> wall -> content -> space_membership -> space.

It's really problematic to just replace a part of it.

**Comment by WASasquatch** [comment]
Hmm seems like a issue with migrating. Especially, if lets say you come from 0.10.0 to a theoretical 2.x.x

Could we work on added a feature to rebuild activity? For example, if there are members, and spaces, add rebuild the system so those members and spaces function without any content in the system by recreating it.

**Comment by WASasquatch** [comment]
Basically it comes down to this

Some people will not update HumHub, some people will only update intermittently.

When these people go to update it's likely HumHub will not know what to do, and break their DBs and HumHub installations. With this in mind, for these instances of broken DBs, or outdated DBs it should ONLY import users and spaces and they SHOULD be generic across the spectrum. 

Same for only uploading only users and spaces to the DB like I have without content so I don't have to remake aaaaaalll of my spaces manually.

Basically, all SPACE information should be contained in the spaces tables, all user information besides LDAP in the User table, so you can always at least migrate users and spaces if the systems posting methods, and other methods dramatically change, or you forget to update throuhg a few versions. 

HumHub is the only software I have come across, that I simply can't migrate database information too because of how weird it is using multiple tables to refer on (which also causes MySQL overhead cross-checking tables constantly)
